src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java

```diff
@@ -18,48 +18,82 @@
 public class BlockContentDisplayView extends ViewPart<WebView> {
 
     private static final String FXML = "blockcontent/BlockContentDisplayView.fxml";
+    private static final String NEWLINE = "\n";
     private final Model model;
 
     @FXML
     private WebView blockContentDisplay;
 
     public BlockContentDisplayView(Model model) {
         super(FXML);
-
         Objects.requireNonNull(model);
-
         this.model = model;
-
         setup();
     }
 
     /**
-     * Sets up the view's initial data and wire up all required change listeners.
+     * Sets up the view's initial data and wires up all required change listeners.
      */
     private void setup() {
-        // Initialize content display
-        reload(model);
-
-        // Setup change listeners
-        model.currentlyOpenPathProperty().addListener(observable -> reload(model));
-        model.getBlockTree().getRootBlock().getTreeItem()
-                .addEventHandler(TreeItem.treeNotificationEvent(), event -> reload(model));
+        setBlockContentDisplay(model);
+        setChangeListeners();
     }
 
     /**
-     * Reloads the {@link WebView}'s content with the currently open block's rendered HTML body.
+     * Updates the content of the view-components to that of the currently open block.
      *
-     * @param model App's model
+     * @param model App's model.
      */
-    private void reload(Model model) {
+    private void setBlockContentDisplay(Model model) {
         Objects.requireNonNull(model);
 
         AbsolutePath currentlyOpenPath = model.getCurrentlyOpenPath();
         BlockTreeItem currentlyOpenBlock = model.getBlockTree().get(currentlyOpenPath);
-        String markdownBody = currentlyOpenBlock.getBody().getText();
 
+        String htmlTitle = getHtmlTitle(currentlyOpenBlock);
+        String htmlBody = getHtmlBody(currentlyOpenBlock);
+        String htmlContent = htmlTitle + NEWLINE + htmlBody;
```

Comment:
Actually, HTML doesn't require respect newlines. You can remove this newline and everything will still work fine!